src/artifactory-client.ts

```diff
@@ -93,20 +93,30 @@ class Artifactory implements ArtifactoryClient {
 		});
 	}
 
-	getContentStream(item: ArtifactoryItemMeta | string): Promise<stream.Readable> {
+	getContentStream(item: ArtifactoryItemMeta | string, rangeHeader?: string): Promise<IncomingMessage> {
 		const uri = this.resolveUri(item);
-		return http.get(uri,
-		{
+
+		const reqData: http.RequestData = {
 			headers: {
 				Authorization: this.authorizationString
 			}
-		});
+		};
+
+		if (rangeHeader && reqData.headers) {
+			reqData.headers.Range = rangeHeader;
+		}
+
+		return http.get(uri, reqData);
 	}
 
 	resolveUri(item: ArtifactoryItemMeta | string): string {
 		const baseUrl = `${this.config.protocol ?? 'https'}://${this.config.host}/artifactory/`;
```

Comment:
why do we need to create `baseUrl` every time when we call this method? Can we move this logic to constructor?

Reply:
and , I think, need to some refactor logic in this method

```suggestion
resolveUri(item: ArtifactoryItemMeta | string): string {
		if (typeof item !== 'string') {
				return `${this.baseUrl}${item.repo}/${item.path}/${item.name}`;
		}
		
		if (item.indexOf(baseUrl) === 0) {
				return item;
		}
		
		return `${this.baseUrl}${item}`;
```